### PR TITLE
[lipstick] Support icons associated with notification remote actions

### PR DESCRIPTION
**Pull request by matthewvogt**
For notifications that contain multiple remote actions, allow an icon
to be associated with each action (as well as a display string).

### CODE REVIEW

File: src/notifications/lipsticknotification.h [hunk 1]
```diff
@@ -46,6 +46,7 @@ class LIPSTICK_EXPORT LipstickNotification : public QObject
     Q_PROPERTY(int priority READ priority NOTIFY priorityChanged)
     Q_PROPERTY(QString category READ category NOTIFY categoryChanged)
     Q_PROPERTY(bool userRemovable READ isUserRemovable NOTIFY userRemovableChanged)
+    Q_PROPERTY(QVariantList remoteActions READ remoteActions)
```
Comment (pvuorela) [inline src/notifications/lipsticknotification.h]: There's setActions() as public method, so technically the property can change too. Though seems actions property and some others are missing signal already.

Comment (matthewvogt) [inline src/notifications/lipsticknotification.h]: Yes, practically speaking there's no possibility of the actions changing - to date there's only been one (remote) action supported.  I could support changing them after creation, but it complicates the manger, the originator and the renderer of the notification, so I would rather have a known use case before doing that.

Comment (giucam) [inline src/notifications/lipsticknotification.h]: Shouldn't the property be marken CONSTANT then? That way there won't be warnings when some qml code binds a property to it.

Comment (matthewvogt) [inline src/notifications/lipsticknotification.h]: Yes, good idea.

Comment (matthewvogt) [inline src/notifications/lipsticknotification.h]: Done.

**Comment by matthewvogt** [comment]
Updated to add improvements to notificationtool.

**Comment by blammit** [comment]
LGTM

**Pull request merged**